app/services/brapi/v1/germplasm_queries.rb

```diff
@@ -122,17 +139,23 @@ def execute_statement(sql, atts)
     # at least until the previous one has been deallocated
     Thread.exclusive do
       @connection.prepare('brapi_statement', sql)
-      
-      if( (atts != nil) && !(atts.empty?) )
-        results = @connection.exec_prepared("brapi_statement", atts)
-      else
-        results = @connection.exec_prepared("brapi_statement")
+      begin
+        if( (atts != nil) && !(atts.empty?) )
+          results = @connection.exec_prepared("brapi_statement", atts)
+        else
+          results = @connection.exec_prepared("brapi_statement")
+        end
+      rescue PG::Error => e
+        @connection.exec("ROLLBACK") 
+        results = nil
+        Rails.logger.warn { "Encountered an error executing a BrAPI germplasm-related query: #{e.message} #{e.backtrace.join("\n")}" }
```

Comment:
I'm retrieving the error notification to the user in the controller when I receive the results = nil : 
if records.nil? || records.size ==0
        render json: { reason: 'Resource not found' }, status: :not_found  # 404
      else

Is it ok? if not, could you give me an example, please?  I was looking for examples in services but I only found things like
...
rescue Exception => e
      message = "Problem accessing public ORCiD API for uid #{uid}."
      message += " Reason: \"#{e.message}\"."
      message += ' No public user information available for the moment.'
      Rails.logger.warn "WARNING: #{message}"
      return { status: :error, message: message }
    end
...
in orcid_client.rb (which I'm not sure if it's generalizable)